### PR TITLE
Laboratory work #4, Varvara Migacheva - 22FPL2

### PR DESCRIPTION
**Pull request by varmig2000**

### CODE REVIEW

File: lab_4_fill_words_by_ngrams/main.py [hunk 1]
```diff
@@ -28,6 +28,16 @@ def _tokenize(self, text: str) -> tuple[str, ...]:  # type: ignore
         Raises:
             ValueError: In case of inappropriate type input argument or if input argument is empty.
         """
+        if not (isinstance(text, str) and text):
+            raise ValueError
```
Comment (artyomtugaryov) [inline lab_4_fill_words_by_ngrams/main.py:32]: add error message

File: lab_4_fill_words_by_ngrams/main.py [hunk 2]
```diff
@@ -28,6 +28,16 @@ def _tokenize(self, text: str) -> tuple[str, ...]:  # type: ignore
...
+
+        tokenized_text = ''
+        for word in text.lower():
+            if word.isalpha() or word.isspace():
+                tokenized_text += word
+            elif word in '?.!':
+                tokenized_text += ' ' + self._end_of_word_token
```
Comment (artyomtugaryov) [inline lab_4_fill_words_by_ngrams/main.py:39]: fstring

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
What about start.py ?

File: lab_4_fill_words_by_ngrams/main.py (hunk 1 above)
Comment (varmig2000) [reply lab_4_fill_words_by_ngrams/main.py]: fixed

File: lab_4_fill_words_by_ngrams/main.py (hunk 2 above)
Comment (varmig2000) [reply lab_4_fill_words_by_ngrams/main.py]: fixed

**Review by artyomtugaryov — COMMENTED** [review]
Good job

**Comment by vitazaytseva** [comment]
Well done! Good job!

**Pull request closed without merging**